### ISSUE TITLE
"Computing explanation" dialog is hanging when asking for explanation in the "Different Individuals" section

### ISSUE THREAD

**Issue by csnyulas** [protegeproject/protege#833]
If I have a simple ontology that has some individuals declared as being different from each other, when I ask for an explanation for why is a certain individual different from the ones listed in the "Different Individuals" section, the "Computing explanations" dialog will stay stuck, even after I close the explanation window. The dialog can't be closed by any means except by closing Protégé. This was observed in Protege 5.5.0-beta-7. It happens both when a reasoner is selected (regardless if it was started/synced or not) and when there is no reasoner selected.

Steps to reproduce:
1. Open Protege
2. In the empty ontology that is shown by default create a class (let's say `A`)
3. Create two individuals of the class created before (let's say `a1` and `a2`)
4. Select one of individual, say `a1`, and click on the "Explain inference" (question mark) icon at the right of the other individual (`a2`) listed in the "Different Individuals" section
5. If an explanation window pops up, close it

Observe that the "Computing explanations" dialog will stay stuck.
 
![image](https://user-images.githubusercontent.com/2441046/51589360-8f08e600-1e9b-11e9-80ae-f975d24e914d.png)

**Comment by rdewaele** [comment]
I observe the same bug on macOS using Protégé 5.5.0 - also when only pressing "stop searching".

This bug is especially obtrusive when working with larger / more complex ontologies. Often I'd like to stop the search because 5 explanations is enough, and finding them all takes $significant_amount minutes.

**Comment by matthewhorridge** [comment]
Possibly related to #879

**Comment by nicolevasilevsky** [comment]
I am having the same problem in MacOS Protege 5.5.0

For some explanations, it is getting stuck for 10+ minutes and won't respond when I click stop searching, I can only quit (or force quit) Protege.

**Comment by rdewaele** [comment]
> I am having the same problem in MacOS Protege 5.5.0
> 
> For some explanations, it is getting stuck for 10+ minutes and won't respond when I click stop searching, I can only quit (or force quit) Protege.
> 
> I am trying to get explanations of subclassOf inferences in mondo-edit.obo.

Given that this issue has the "needs reproducing" tag, could you be a bit more specific in what ontology you have loaded, and which inference you are trying to get explained exactly?
If it is public data (mine sadly is not), I can try it on my machine as well, and if I have the same behaviour, chances are that Matthew will be able to reproduce it as well.

**Comment by nicolevasilevsky** [comment]
@rdewaele Yes - the ontology I am using is mondo-edit.obo, which you can download here:
https://github.com/monarch-initiative/mondo

An example of an inference I am trying to get explained is the subclassOf inference for AICA-ribosiduria (MONDO:0012099)

**Comment by jbulow** [comment]
Same issue on ubuntu 18.04.
